modules/test/playwright/tests/object-web/rootModels.spec.ts

```diff
@@ -399,11 +399,12 @@ test.describe('Manage root models elements through Objects Admin', () => {
 	});
 
 	test('can create relationship with inheritance enabled using the add relationship modal', async ({
+		addNewObjectRelationshipModalPage,
```

Comment:
Yeah, I know... here is what happened:

locators in `objectRelationshipFormPage` can fail because of the container element in which the form is. For example:
```
this.labelInput = page.getByLabel('LabelMandatory');
```
There may be other elements in the dom that match this locator, so we need to scope it like
```
this.labelInput = page.locator('.modal-content').getByLabel('LabelMandatory');
```
This means that locators in `objectRelationshipFormPage` need to be initialized with a scope to work properly, specially if you consider that the same form can be found in model builder sidebar, admin sidepanel and the add new relationship modal.

Ultimately, this leads to a situation where `objectRelationshipFormPage` needs more than page to be initialized, like this:
```
this.objectRelationshipFormPage = new ObjectRelationshipFormPage(
	page,
	'.modal-content'
);
```
And this is the reason why we should never use `objectRelationshipFormPage` directly. We will always be calling it through the container page, which already initialized it correctly. The instance of object `objectRelationshipFormPage` that we get in line 407 was already initialized with `'.modal-content'` in `addNewObjectRelationshipModalPage`, this is why we get from there. The same will happen when we create pages like `editObjectRelationshipSidepanelPage`, which will do its own initialization of `objectRelationshipFormPage` with the right container selector.

Reply:
Thanks for the explanation!